### DISCUSSION TITLE
createResource not updating when using stores

### DISCUSSION THREAD

**Discussion by FallingSnow** [solidjs/solid#902] [Q&A] [answered]
When I update `controls.window` via `setControls('window', 100)` the value in the dom does update to reflect this change, but the resource is not queried again with the new value. I'm sure I'm doing something wrong, but I can't quite figure out what it is. I've tried passing the whole controls variable as well.
```js
const fetchData = async ({window}) => {
  console.log('loading data')
  const end = Date.now();
  const start = end - window * 1000;
  console.debug({ end, start, window });
  let res = await fetch(`https://localhost:8081/api/data?symbol=BTC&start=${start}&end=${end}`);

  let root = new Uint8Array(await res.arrayBuffer());
  let buf = new flatbuffers.ByteBuffer(root);
  let data = Data.getRootAsData(buf);
  return data;
}

function App() {
  const [controls, setControls] = createStore({
    window: 604_800,
    bucketSize: 900
  });
  const [data] = createResource({window: controls.window}, fetchData);

  return (
    <div class={styles.App}>
      <header class={styles.header}>
        <ControlBar controls={controls} onChange={setControls} />
        <Graph data={data} />
        {controls.window}
      </header>
    </div>
  );
}

export default App;
```

**Comment by ryansolid** [comment, marked as answer]
Try:
```js
const [data] = createResource(() => ({window: controls.window}), fetchData);
```
It needs to be wrapped in a function to track the expression.

**Reply by FallingSnow** [reply]
Wow, that worked. Thank you.

Small fix: The return value needed to be wrapped in parentheses:
```js
const [data] = createResource(() => ({window: controls.window}), fetchData);
```

**Reply by ryansolid** [reply]
Yes.. right otherwise it assumes the curly braces are part of the function body. I'm glad it worked for you.

**Reply by juniorforlife** [reply]
@ryansolid why does it have to be a function in this case and not with signals?
I've been trying Solid for the first time and I really like its reactivity where you don't have to re-render everything because a piece of state changes.
However I find it really confusing and inconsistent with how to use the reactivity system specifically with store and signal.
I bet a lot of beginners will be frustrated not knowing why their reactivity doesn't work because they:
1. forget to call the signal getter
2. destructure store value instead of accessing it directly
3. use a store value directly instead of putting it in a function like the discussion above

It would be great if we can improve the apis. If not then please have a gotchas section to consolidate all the common mistakes beginner makes and add linting warning when a mistake is made.

**Reply by snnsnn** [reply]
You need to read the docs, it is short and addresses all your concerns along with the reasons behind them.
